pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -115,8 +127,12 @@ type Registry struct {
 	Blocked bool `toml:"blocked,omitempty"`
 	// If true, mirrors will only be used for digest pulls. Pulling images by
 	// tag can potentially yield different images, depending on which endpoint
-	// we pull from.  Forcing digest-pulls for mirrors avoids that issue.
+	// we pull from.  Restricting mirrors to pulls for mirrors avoids that issue.
 	MirrorByDigestOnly bool `toml:"mirror-by-digest-only,omitempty"`
+	// If true, mirrors will only be used for tag pulls. Pulling images by
+	// tag can potentially yield different images, depending on which endpoint
+	// we pull from.  Restricting mirrors to pulls for mirrors avoids that issue.
+	MirrorByTagOnly bool `toml:"mirror-by-tag-only,omitempty"`
```

Comment:
… Do we need this setting at all? I don’t really see why one would ever enable this. AFAIK, tag-only mirrors only make sense if there are other digest-only mirrors.

@vrothberg WDYT?

Reply:
IIRC, I was asking to add it but I am not feeling strongly.  _But_ I am worried about the ambiguity of a mixed list of mirrors. Let's consider the following:
* Mirror A: digest-only=true
* Mirror B: digest-only=false
* Mirror C: digest-only=true

Now, I am pulling `foo@digest`.  Would we attempt a pull from `Mirror B`?  `digest-only=false` would intuitively imply that I can safely pull a digest from it but in the context of the other mirrors it does not really makes sense.

Maybe `mirror-by-tag-only` should only be set internally if a "sibling mirror" sets `digest-only=true`?


Making configuration settings user-friendly is hard.

Reply:
My mind is sufficiently twisted by programming that “`digest-only = false` allows working with digests” feels obvious and intuitive to me, but I know how I am at UI…

---

How about reformulating this as:
- `pull-from-mirror=all[-references]` (the default)
- `pull-from-mirror=digest-only`
- `pull-from-mirror=tag-only`

That’s the same set of choices, but 1) it is impossible to set digest-only and tag-only at the same code, avoiding a corner case, and 2) it might make it clearer what the default is (and 3) it allows us to define a different set of mirrors for pushing, if we ever added that feature).

Reply:
> How about reformulating this as:

That is by far the cleanest version IMHO. @QiWang19, what do you think?